### PR TITLE
Fix gh docker url

### PR DESCRIPTION
**Pull request by uuf6429**
I've found and solved some different problems in this PR:
- Something weird happens to docker-compose when run in the background. In particular, it doesn't bind exposed ports to 0.0.0.0/localhost anymore (main problem and motivation behind this PR)
- We run docker-compose up before installing dependencies, this is bad because we are also sharing a dependency with the selenium container (this is probably what caused permission problems when installing dependencies).
- selenium api url was apparently wrong (causing http 403 forbidden) it needs to have the `wd/hub` path

### CODE REVIEW

**Comment by uuf6429** [comment]
@stof any idea why the workflows are failing now? it was working a few moments ago and I can't see how my changes could have caused it.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -68,14 +63,18 @@ jobs:
...
           composer update --no-interaction --prefer-dist --ansi --no-progress
 
+      - name: Start Selenium
+        run: |
+          SELENIUM_IMAGE=selenium/standalone-${{ matrix.browser }}:${{ matrix.selenium }} docker compose up --wait
+
       - name: Wait for selenium to start
         run: |
-          curl --head -X GET --silent --show-error --retry 60 --retry-connrefused --retry-delay 1 --max-time 10 http://172.18.0.2:4444
+          curl --retry 5 --retry-all-errors --retry-delay 1 --max-time 10 --head -X GET http://localhost:4444/wd/hub/status
```
Comment (stof) [inline .github/workflows/ci.yml:72]: `--head -X GET` does not make sense to me. We cannot do both a HEAD and a GET request.

Comment (uuf6429) [reply .github/workflows/ci.yml:72]: yeah weird.. copy & paste mistake, `-X GET` especially doesn't make sense (it's the default anyway).

Comment (uuf6429) [reply .github/workflows/ci.yml:72]: @stoff actually apparently it makes sense, see here: https://stackoverflow.com/a/63867207/314056

That causes curl to do a GET request (`-X GET`) but drop the body (`--head`), whereas `--head` by itself does a HEAD request (and also ignores the body).

But maybe it actually makes sense for us to actually display the body?

**Pull request merged**